src/openrct2-ui/UiContext.Linux.cpp

```diff
@@ -113,16 +115,72 @@ namespace OpenRCT2::Ui
             }
         }
 
+        int ShowMenuDialog(const std::vector<std::string>& options, const std::string& title, const std::string& text) override
+        {
+            std::string executablePath;
+            DIALOG_TYPE dtype = GetDialogApp(&executablePath);
+
+            size_t longest_string = 0;
+            for(const auto& option : options) {
+                if(option.size() > longest_string) {
+                    longest_string = option.size();
+                }
+            }
+
+            // If zenity and kdialog were sane, they would support some kind of
```

Comment:
Avoid putting opinions in the code. Just state they do not support automatic scaling and leave it at that.